ttg/ttg/parsec/ttg.h

```diff
@@ -606,7 +606,7 @@ namespace ttg_parsec {
         }
 
         // We store the hash of the key and the address where it can be found in locals considered as a scratchpad
-        uint64_t hv = ttg::hash<decltype(key)>{}(key);
+        uint64_t hv = ttg::hash<std::decay_t<decltype(key)>>{}(key);
```

Comment:
@devreal seems like we hashed refs to keys rather than keys themselves here ... not sure how this worked